impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -311,6 +320,29 @@ std::optional<Entity> GaussianBlurFilterContents::RenderFilter(
   Vector2 pass1_pixel_size =
       1.0 / Vector2(pass1_out.value().GetRenderTargetTexture()->GetSize());
 
+  std::optional<Rect> input_snapshot_coverage = input_snapshot->GetCoverage();
+  Quad blur_uvs = {Point(0, 0), Point(1, 0), Point(0, 1), Point(1, 1)};
+  if (expanded_coverage_hint.has_value() &&
+      input_snapshot_coverage.has_value() &&
+      // TODO(tbd): Remove this condition. There is some flaw in coverage
+      //            stopping us from using this today. I attempted to use source
+      //            coordinates to calculate the uvs, but that didn't work
+      //            either
+      //            (https://github.com/flutter/engine/pull/49299#issuecomment-1865311438).
+      input_snapshot.has_value() &&
+      input_snapshot.value().transform.IsTranslationScaleOnly()) {
```

Comment:
I think this needs to:
* check the `entity.GetTransform()` (rather than the `input_snapshot`'s transform)
* check if the transform is translation-only (rather than scale-translation-only)

If either one of these isn't true, then I think `coverage_hint` and `input_snapshot_coverage` could end up being in difference spaces in some situations, which will cause a variation of the UV mapping bug.

Reply:
If you're up for it, one way you could smoke test this issue is by adding a scale param to the interactive Aiks toy.

Reply:
I added the parameter to the interactive test and added an automated test.  Let me know if you are thinking of some other case where this may break.

Reply:
To be clear it's necessarily that I don't believe there can be a problem; I just don't want to shrink the domain of this optimization without proof that it doesn't work everywhere via a test.  That test can serve as the basis for expanding the domain in the future.

Reply:
Oh yeah, I get you. Also I think you should land this at will, we can patch other issues as we find em. ;)

It would also be good to test a case where the transform+coverage rect is absorbed by the snapshot rather than forwarded. For example: A blur applied to a gradient circle (can't be a solid color since that'll fall into the rrect fast path).

Reply:
SGTM